eZ/Publish/API/Repository/Values/Content/Query/Aggregation/FieldAggregationInterface.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository\Values\Content\Query\Aggregation;
+
+interface FieldAggregationInterface
```

Comment:
Please keep the naming convention which has been used by [the Repository API designers](https://github.com/ezsystems/ezplatform-kernel/tree/v1.1.0/eZ/Publish/API/Repository), at least for this package. Interface already knows it's an interface. Other variants chosen in Kernel are mistakes, done in haste, w/o proper code review, I presume.

Implementations should have distinct keyword related to their function (I'm also against suffixing them with "impl") and can be internal so it's quite visible if someone makes a mistake.

PSR "Interface" thing is not domain driven (it's an opposition of DDD naming) and it's their [by-law](https://www.php-fig.org/bylaws/psr-naming-conventions/), not an actual PSR.

Good summary of why what we're trying to do here is wrong is [here](https://www.alainschlesser.com/interface-naming-conventions/). Again, this is not a convention based on this post, but a summary of why Repository API is designed in this way, not the other one. Source of this comes from DDD by E. Evans, not blogs, nor Reddit, nor SO.

Reply:
Besides my previous comments on this topic, this class is in the `Values` namespace. Without this suffix, it will be hard to guess that this is an interface. Your arguments about using FQCN in docblock applies here

Reply:
**I decided to remove "Interface" suffix and stay consistent with other API classes.** We will back to this topic on Slack/Zoom after 3.2 release.

Reply:
> Your arguments about using FQCN in docblock applies here (about readability during code review). 

Weren't you against that as well?

> For me, it is easier to read and understand.

I can find examples multiple linguistic mistakes and errors which are easier to read and understand, but nonetheless are still mistakes and errors.